espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/adapter/espresso/EspressoW3CActionAdapter.java

```diff
@@ -342,19 +348,22 @@ public int getKeyCode(String keyValue, int location) throws AppiumException {
     }
 
     public long getViewportHeight() {
-        // Stub.
-        return Long.MAX_VALUE;
+        return displayMetrics.heightPixels;
     }
 
     public long getViewportWidth() {
-        // Stub.
-        return Long.MAX_VALUE;
+        return displayMetrics.widthPixels;
     }
 
     public long[] getElementCenterPoint(String elementId)
             throws NoSuchElementException, StaleElementException, NotYetImplementedException {
-        // Stub.
-        return new long[] { };
+        View view = Element.getViewById(elementId);
+        int[] out = new int[2];
+        view.getLocationInWindow(out);
+        long[] res = new long[2];
+        res[0] = out[0] + ((view.getWidth()) / 2);
```

Comment:
+=

Reply:
res[0] and out[0] are different

Reply:
ah, missed that

Reply:
In general I would rather prefer to use an object, like Point to keep the coordinates rather than an array. It is not C ;)

Reply:
You mean for the return value?

Reply:
yes